### PR TITLE
Expose methods for related links

### PR DESCRIPTION
**Pull request by tijmenb**
This PR adds a method on the helper to generate the payload for the "related items" component:

http://govuk-component-guide.herokuapp.com/components/related_items

It introduces a couple of new classes. After this is merged we can also use the `ContentItem` class in the breadcrumbs.

https://trello.com/c/gXQ9fdGe

### CODE REVIEW

**Comment by MatMoore** [comment]
I added usage instructions (using breadcrumbs as an example) to https://github.com/alphagov/govuk_navigation_helpers/pull/3

And yardoc in https://github.com/alphagov/govuk_navigation_helpers/pull/7

**Comment by MatMoore** [comment]
`ordered_related_items` works for me 👍

**Comment by tijmenb** [comment]
Tests will pass after https://github.com/alphagov/govuk-content-schemas/pull/401

File: lib/govuk_navigation_helpers.rb [hunk 1]
```diff
@@ -14,6 +15,13 @@ def breadcrumbs
       Breadcrumbs.new(content_item).breadcrumbs
     end
 
+    # Generate a the related items payload
```
Comment (carvil) [inline lib/govuk_navigation_helpers.rb]: Typo, should be `Generate the...` (without the `a`).

File: lib/govuk_navigation_helpers/grouped_related_links.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+      # as the content item.
+      :related_with_parent_in_common,
+
+      # This will contain related items that have a grandparent in common with
+      # the content item.
+      :related_with_grandparent_in_common,
+
+      # This will contain the related items that have a completely different
+      # parent/breadcrumb from the content item.
+      :related_with_nothing_in_common
```
Comment (carvil) [inline lib/govuk_navigation_helpers/grouped_related_links.rb]: This name is a bit confusing to me. If they have nothing in common, how are they related? Would it make sense to rename this field?

File: lib/govuk_navigation_helpers/grouped_related_links.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      @related_with_grandparent_in_common = []
+      @related_with_nothing_in_common = []
+      group_related_links!
+    end
+
+  private
+
+    def group_related_links!
+      content_item.related_links.each do |related_item|
+        if content_item.parent && related_item.parent
```
Comment (carvil) [inline lib/govuk_navigation_helpers/grouped_related_links.rb]: What do you think of refactoring all these `if`, `else`, `elsif` clauses? I personally find it a bit hard to follow. It could be a bit more verbose, something like this:

```
def related_with_parent_in_common
  @related_with_parent_in_common ||=
    content_item.related_links.select { |related_item| related_parent?(related_item) }
end

def related_with_grandparent_in_common
  @related_with_grandparent_in_common ||=
    content_item.related_links.select { |related_item| related_grandparent?(related_item) }
end

def related_with_nothing_in_common
  @related_with_nothing_in_common ||=
    content_item.related_links.select { |related_item| unrelated?(related_item) }
end

private

def related_parent?(related_item)
  ...
end

...
```

What do you think?

**Comment by carvil** [comment]
:shipit:
